### ISSUE TITLE
Plugin with id 'maven' not found for `:unimodules-core`  and compileSdkVersion is not specified

### ISSUE THREAD

**Issue by ajitdas123** [expo/expo#18715]
### Summary

We are using bare minimum expo on our react native project which was created by using `react-native-cli` . Inorder to use some expo module we had to install the expo client as detailed in the documentation here 
https://docs.expo.dev/bare/installing-expo-modules/
The installation was completed  but when trying to build the app on android, its throwing error saying 
```
* What went wrong:
A problem occurred evaluating project ':unimodules-core'.
> Plugin with id 'maven' not found.

2: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':expo'.
> compileSdkVersion is not specified. Please add it to build.gradle

```
 Found a similar issue but it doesn't have any answer https://github.com/expo/expo/issues/16173
 

### What platform(s) does this occur on?

Android

### Environment

Here is the information on the packages

```
    "expo": "^46.0.8",
    "expo-gl": "^11.4.0",
    "react": "18.0.0",
    "react-native": "0.69.3"
```

### Minimal reproducible example

The application was created using `react-native-cli` . All we did is run 

`npx install-expo-modules@latest`  

As mentioned in the official documentation https://docs.expo.dev/bare/installing-expo-modules/

**Comment by Kudo** [comment]
hi there! the `unimodules-core` is deprecated for a while. could you double check why this module was being included, maybe through `yarn why @unimodules/core`.
otherwise, please try to create a github repo with minimal reproducible example for us to have further investigation. thanks!

**Comment by ajitdas123** [comment]
@Kudo Yes, i will try to do that. Ty

**Comment by jester-twoconnect** [comment]
has there any workaround from this issue i'm experiencing it as well after i upgrade for expo version 46.0.0

**Comment by zabsalahid** [comment]
me too, expo version 46.0.0

```
[stderr] FAILURE: Build completed with 2 failures.
[stderr] 1: Task failed with an exception.
[stderr] -----------
[stderr] * Where:
[stderr] Build file '/home/expo/workingdir/build/node_modules/@unimodules/core/android/build.gradle' line: 3
[stderr] * What went wrong:
[stderr] A problem occurred evaluating project ':unimodules-core'.
[stderr] > Plugin with id 'maven' not found.
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] 2: Task failed with an exception.
[stderr] -----------
[stderr] * What went wrong:
[stderr] A problem occurred configuring project ':expo'.
[stderr] >
[stderr] compileSdkVersion is not specified. Please add it to build.gradle
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 2m 3s
```

**Comment by huifng** [comment]
The maven plugin was removed, instead by maven-publish. https://docs.gradle.org/7.0/userguide/publishing_maven.html#publishing_maven. I followed the doc and replaced all modules, the issue was addressed.

**Comment by bianbianzhu** [comment]
> The maven plugin was removed, instead by maven-publish. https://docs.gradle.org/7.0/userguide/publishing_maven.html#publishing_maven. I followed the doc and replaced all modules, the issue was addressed.

Could you plz explain a bit more? Like, do we need to go to each of the node modules and replace something? Thanks in advance.

**Comment by lanceliamll** [comment]
Does this have fix yet? Im currently encountering this issue

**Comment by akhilv77** [comment]
> me too, expo version 46.0.0 anyone managed to have a workaround on this?
> 
> ```
> [stderr] FAILURE: Build completed with 2 failures.
> [stderr] 1: Task failed with an exception.
> [stderr] -----------
> [stderr] * Where:
> [stderr] Build file '/home/expo/workingdir/build/node_modules/@unimodules/core/android/build.gradle' line: 3
> [stderr] * What went wrong:
> [stderr] A problem occurred evaluating project ':unimodules-core'.
> [stderr] > Plugin with id 'maven' not found.
> [stderr] * Try:
> [stderr] > Run with --stacktrace option to get the stack trace.
> [stderr] > Run with --info or --debug option to get more log output.
> [stderr] > Run with --scan to get full insights.
> [stderr] ==============================================================================
> [stderr] 2: Task failed with an exception.
> [stderr] -----------
> [stderr] * What went wrong:
> [stderr] A problem occurred configuring project ':expo'.
> [stderr] >
> [stderr] compileSdkVersion is not specified. Please add it to build.gradle
> [stderr] * Try:
> [stderr] > Run with --stacktrace option to get the stack trace.
> [stderr] > Run with --info or --debug option to get more log output.
> [stderr] > Run with --scan to get full insights.
> [stderr] ==============================================================================
> [stderr] * Get more help at https://help.gradle.org
> [stderr] BUILD FAILED in 2m 3s
> ```

I am getting the same issue, do you find any workaround for this ?

**Comment by huifng** [comment]
We finally found the root cause is from 3rd dependencies. You need to check all your dependencies, maybe they are depending on the old expo version, which caused the 'maven' plugin. if it's the case, you have to upgrade the dependencies to the latest expo version as well.

**Comment by akhilv77** [comment]
Package.json



{
  "name": "new-expo-app",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "build": "eas build --platform android",
    "build-web": "expo export:web"
  },
  "dependencies": {
    "@aws-amplify/core": "^5.0.4",
    "@pixi/constants": "^7.0.4",
    "@pixi/core": "^7.0.4",
    "@pixi/filter-alpha": "^7.0.4",
    "@pixi/filter-blur": "^7.0.4",
    "@react-native-async-storage/async-storage": "^1.17.11",
    "@react-native-community/datetimepicker": "6.5.2",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "9.3.5",
    "@react-navigation/compat": "^5.3.20",
    "@react-navigation/drawer": "^6.5.3",
    "@react-navigation/native": "^6.0.16",
    "@react-navigation/stack": "^6.3.7",
    "aws-amplify": "^5.0.4",
    "axios": "^1.2.1",
    "deprecated-react-native-listview": "^0.0.8",
    "expo": "~47.0.8",
    "expo-asset": "^8.6.2",
    "expo-build-properties": "^0.4.1",
    "expo-constants": "^14.0.2",
    "expo-document-picker": "^11.0.1",
    "expo-file-system": "^15.1.1",
    "expo-font": "^11.0.1",
    "expo-gl": "^12.0.1",
    "expo-image-picker": "^14.0.2",
    "expo-linear-gradient": "^12.0.1",
    "expo-linking": "^3.2.3",
    "expo-modules-core": "^1.0.3",
    "expo-permissions": "^14.0.0",
    "expo-pixi": "^2.1.1",
    "expo-splash-screen": "^0.17.5",
    "expo-status-bar": "~1.4.2",
    "fiction-expo-restart": "^1.1.3",
    "galio-framework": "^0.8.0",
    "moment": "^2.29.4",
    "react": "18.1.0",
    "react-dom": "^18.1.0",
    "react-native": "0.70.5",
    "react-native-calendar-picker": "^7.1.3",
    "react-native-gesture-handler": "^2.8.0",
    "react-native-gifted-chat": "^1.0.4",
    "react-native-image-modal": "^2.0.4",
    "react-native-image-to-pdf": "^1.2.0",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-masked-text": "^1.13.0",
    "react-native-modal-dropdown": "^1.0.2",
    "react-native-paper": "^4.12.5",
    "react-native-reanimated": "~2.12.0",
    "react-native-safe-area-context": "^4.4.1",
    "react-native-screens": "^3.18.2",
    "react-native-web": "^0.18.10",
    "react-native-webview": "11.23.1",
    "rn-pdf-reader-js": "4.1.1"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "cross-env": "^7.0.3",
    "jest-expo": "^47.0.1"
  },
  "private": true
}

**Comment by PolpEdu** [comment]
same issue here with package.json:
`
{
  "name": "appsportnature",
  "version": "3.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "tunnel": "expo start --tunnel",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject",
    "eas-build-post-install": "echo 456",
    "eas-build-on-success": "echo 789",
    "eas-build-on-error": "echo 012"
  },
  "dependencies": {
    "@expo-google-fonts/dm-sans": "^0.2.2",
    "@expo-google-fonts/inter": "^0.2.2",
    "@react-native-async-storage/async-storage": "~1.17.3",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-picker/picker": "2.4.8",
    "@react-navigation/bottom-tabs": "^6.3.1",
    "@react-navigation/native": "^6.0.10",
    "@react-navigation/native-stack": "^6.5.2",
    "@react-navigation/stack": "^6.2.1",
    "@tmcw/togeojson": "^5.2.2",
    "@unimodules/core": "^7.2.0",
    "axios": "^0.26.1",
    "bcrypt": "^5.0.1",
    "bcryptjs": "^2.4.3",
    "buffer": "^6.0.3",
    "expo": "^47.0.0",
    "expo-app-loading": "~2.1.1",
    "expo-application": "~5.0.1",
    "expo-auth-session": "~3.7.3",
    "expo-av": "~13.0.2",
    "expo-background-fetch": "~11.0.1",
    "expo-barcode-scanner": "~12.1.0",
    "expo-camera": "~13.1.0",
    "expo-device": "~5.0.0",
    "expo-facebook": "~12.2.0",
    "expo-file-system": "~15.1.1",
    "expo-font": "~11.0.1",
    "expo-google-app-auth": "~8.3.0",
    "expo-google-sign-in": "~10.2.0",
    "expo-location": "~15.0.1",
    "expo-media-library": "~15.0.0",
    "expo-modules-autolinking": "~1.0.0",
    "expo-notifications": "~0.17.0",
    "expo-random": "~13.0.0",
    "expo-status-bar": "~1.4.2",
    "expo-storage": "^44.0.0",
    "expo-task-manager": "~11.0.1",
    "expo-updates": "~0.15.6",
    "fbemitter": "^3.0.0",
    "fiction-expo-restart": "^1.0.5",
    "geolib": "^3.3.3",
    "google-auth-library": "^8.0.1",
    "i18next": "^21.8.9",
    "i18next-browser-languagedetector": "^6.1.4",
    "i18next-http-backend": "^1.4.1",
    "lodash": "^4.17.21",
    "moment": "^2.29.2",
    "prop-types": "^15.8.1",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-i18next": "^11.17.1",
    "react-native": "0.70.5",
    "react-native-draggable-flatlist": "^3.1.2",
    "react-native-dropdown-picker": "^5.4.2",
    "react-native-elements": "^3.4.2",
    "react-native-geojson": "^0.1.1",
    "react-native-gesture-handler": "~2.8.0",
    "react-native-maps": "1.3.2",
    "react-native-modals": "^0.22.3",
    "react-native-paper": "^4.12.2",
    "react-native-photo-zoom": "^1.1.5",
    "react-native-popup-dialog": "^0.18.3",
    "react-native-push-notification": "^8.1.1",
    "react-native-reanimated": "~2.12.0",
    "react-native-responsive-fontsize": "^0.5.1",
    "react-native-restart": "^0.0.24",
    "react-native-safe-area-context": "4.4.1",
    "react-native-screens": "~3.18.0",
    "react-native-snap-carousel": "^3.9.1",
    "react-native-sound-player": "^0.12.1",
    "react-native-svg": "13.4.0",
    "react-native-svg-transformer": "^1.0.0",
    "react-navigation": "^4.4.4",
    "react-navigation-stack": "^2.10.4",
    "react-svg-text": "^0.1.2",
    "regenerate": "^1.4.2",
    "regjsgen": "^0.7.1",
    "xmldom": "^0.6.0"
  },
  "devDependencies": {
    "@babel/core": "^7.19.3",
    "@babel/preset-react": "^7.18.6",
    "@types/lodash": "^4.14.181",
    "@types/react": "~18.0.24",
    "@types/react-dom": "~18.0.8",
    "@types/react-native": "~0.70.6",
    "@types/react-native-popup-dialog": "^0.16.4",
    "@types/react-native-snap-carousel": "^3.8.5",
    "babel-plugin-module-resolver": "^4.1.0",
    "deprecated-react-native-prop-types": "^3.0.1",
    "typescript": "^4.6.3"
  },
  "resolutions": {
    "@types/react": "^18.0.8",
    "@types/react-dom": "^18.0.5",
    "@types/react-native": "~0.69.1",
    "graphql": "^16.5.0"
  },
  "private": true
}
`

**Comment by PolpEdu** [comment]
FOUND IT GUYS:
ok first i reverted back to expo 45, then i had a module called "fiction-expo-restart" that depended on a unimodules module, i substituted it for `{ Update } from "expo"` in the code and it managed to build with `aes build -p android` ! YES

**Comment by PolpEdu** [comment]
TL;DR review your package-lock.json for a unimodules dependency and delete it.

**Comment by thanhdat2106** [comment]
same issue

**Comment by muhammadasad22** [comment]
The Only solution of this problem is that ::
Replace `apply plugin: 'maven'` to `apply plugin: 'maven-publish'` in node_modules>expo package> android > build.gradle
Also Replace 
```
// uploadArchives {
//   repositories {
//     mavenDeployer {
//       configuration = configurations.deployerJars
//       repository(url: mavenLocal().url)
//     }
//   }
// }
```
to 

```
publishing {
    repositories {
        maven {
            url = uri("${rootProject.projectDir}/maven-repo")
        }
    }
}
```
For more details visit this doc :[]( https://docs.gradle.org/7.0/userguide/publishing_maven.html#publishing_maven)
